src/number-formatting/index.ts

```diff
@@ -111,28 +122,30 @@ class NumberFormatting implements NumberFormattingOptions {
             num,
             Math.min(
                 maxDecimals,
-                Math.max(minDecimals, this.getActualDecimals(num)),
+                Math.max(minDecimals, this.getActualDecimals(Number(num))),
             ),
             this,
         );
     }
 
     /**
      * Formats a number into a short format, e.g. 10.000 becomes 10k.
-     * @param number - number
+     * @param number - number to format
      *
      */
-    shortFormat(number: number) {
+    shortFormat(number: number | string | null | undefined) {
+        if (number === undefined || number === null) {
+            return '';
+        }
         return shortFormat(number, this);
     }
 
     /**
      * Returns the actual number of decimals that a number has.
-     * @param number - number
-     *
+     * @param number - number or numeric string
      */
     getActualDecimals(number: number) {
-        number = Math.abs(number);
+        number = Math.abs(Number(number));

```

Comment:
Do you need to do it here? I see it being done on the callsites too..

Reply:
Ahh, yeah, thanks, I allowed strings before so it's not needed now.